### ISSUE TITLE
evaluation in TFTrainer does not run on GPU

### ISSUE THREAD

**Issue by rohanshingade** [huggingface/transformers#11590]
## Environment info

- `transformers` version: 4.5.1
- Platform: ubuntu 18.04
- Python version: 3.6.9
- PyTorch version (GPU?):
- Tensorflow version (GPU?): tensorflow-gpu==2.4.1
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help

@patil-suraj @Rocketknight1

## Information

I'm using TFT5ForConditionalGeneration for masked language modelling task. During training GPU utilisation is above 95% but as soon as evaluation starts it goes to 0%. Evaluation is slow. Even though [evaluate function is in strategy.scope()](https://github.com/huggingface/transformers/blob/c065025c4720a87783ca504a9018454893d00649/src/transformers/trainer_tf.py#L580). it does not use gpu.

The problem arises when using:
* [ ] the official example scripts: (give details below)

I'm using the official example script of TFTrainer and modified `run_tf_glue.py` a bit for custom data input.

The tasks I am working on is:
* [ ] my own task or dataset: (give details below)

Final train_dataset and eval_dataset (input to TFTrainer) have the form `({"input_ids": , "attention_mask": ,"decoder_attention_mask": }, labels)`

## To reproduce

I tried reproducing the error using run_tf_squad.py and run_tf_glue.py but both the scripts gave error as the inputs to the trainer were not compatible. Only MRPC task worked, but it had only 400 examples in evaluation so hard to determine. Rest of them simply didn't work, there was an error.

If possible I would like to contribute to TFTrainer in terms of running evaluation on GPU and processing squad and glue dataset to match dimensions to TFTrainer inputs. Guidance is really appreciated.

**Comment by Rocketknight1** [comment]
Hi, TF maintainer here! We're currently in the process of rewriting the examples. We're deprecating TFTrainer and using more native Keras. Rewriting our GLUE examples is coming up very soon on my to-do list.

That said, the `run_text_classification.py` script is updated to our new standards - feel free to try that and just adapt the input data to use GLUE instead of your own inputs. If that doesn't work for you, I'll try to get the real GLUE script done soon!

**Comment by mymusise** [comment]
Hello, @Rocketknight1, great job, looking forward to the new version of the real GLUE script.

I found `TFTrainer` is still used in the `run_text_classification.py`, does `TFTrainer` will be abandoned?

**Comment by Rocketknight1** [comment]
Are you sure? I can't see it anywhere: https://github.com/huggingface/transformers/blob/master/examples/tensorflow/text-classification/run_text_classification.py

**Comment by mymusise** [comment]
Sorry, you're right! I got another file here: https://github.com/huggingface/transformers/blob/master/examples/legacy/text-classification/run_tf_text_classification.py

Thank you, I will refer to it.

**Comment by ofersabo** [comment]
Hi,

Do you have any estimation as to when the new examples will be ready? 

We're at a starting point of a project in which we want to finetune a pre-trained model in TF 2.3.1, we were thinking to use the HF TFTrainer. 
Given that TFTrainer goes back to use Keras we are interested in the integration between TFTrainer and Keras. 

Thanks for the information
